src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -48,25 +49,27 @@
 
 (s/defn ^:private fingerprint-table!
   [table :- i/TableInstance, fields :- [i/FieldInstance]]
-  (transduce identity
-             (redux/post-complete
-              (f/fingerprint-fields fields)
-              (fn [fingerprints]
-                (reduce (fn [count-info [field fingerprint]]
-                          (cond
-                            (instance? Throwable fingerprint)
-                            (update count-info :failed-fingerprints inc)
-
-                            (some-> fingerprint :global :distinct-count zero?)
-                            (update count-info :no-data-fingerprints inc)
-
-                            :else
-                            (do
-                              (save-fingerprint! field fingerprint)
-                              (update count-info :updated-fingerprints inc))))
-                        (empty-stats-map (count fingerprints))
-                        (map vector fields fingerprints))))
-             (metadata-queries/table-rows-sample table fields {:truncation-size truncation-size})))
+  (let [options (when (driver/supports? driver/*driver* :expressions)
```

Comment:
I think a better place for this would be `table-rows-sample`. There's no need for `fingerprint-table!` to be aware of (and coupled to) implementation details and limitations of the sampling method it uses.

Reply:
yeah i went back and forth on that. wasn't sure who should be concerned with that. the caller asking for truncation or the implementation. felt a bit weird that you could ask for truncation at the call site but then not get it. felt like a bit of a surprise. but i'll move it

Reply:
gonna wait for this test pass to finish before pushing that up

Reply:
That's a fair point. Pick whichever you think makes more sense.